File: src/AutoRest.SdkExplorer/Model/OpenAi/AiChatContext.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;
using AutoRest.SdkExplorer.Model.Code;

namespace AutoRest.SdkExplorer.Model.OpenAi
{
    public class AiChatContext
    {
        public List<AiChatMessage> ChatMessages { get; set; } = new List<AiChatMessage>();
        public List<ApiObjectPath> FunctionPaths { get; set; } = new List<ApiObjectPath>();

        public int? Seed { get; set; }
        public float? NucleusSamplingFactor { get; set; }
        public float? PresencePenalty { get; set; }
        public float? FrequencyPenalty { get; set; }
        public float? Temperature { get; set; }

        public AiChatContext()
        {
        }
    }
}

File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Ai/AiChatContext.ts
import { ApiObjectPath } from "../Code/ApiObjectPath";
import { AiChatMessage } from "./AiChatMessage";

export class AiChatContext implements AutoRest.SdkExplorer.Interface.AiChatContext {
    chatMessages?: AiChatMessage[];
    functionPaths?: ApiObjectPath[];

    seed?: number;
    nucleusSamplingFactor?: number;
    presencePenalty?: number;
    frequencyPenalty?: number;
    temperature?: number;

    constructor(data: AutoRest.SdkExplorer.Interface.AiChatContext) {
        this.chatMessages = data.chatMessages?.map(m => new AiChatMessage(m));
        this.functionPaths = data.functionPaths?.map(p => new ApiObjectPath(p));
        this.seed = data.seed;
        this.nucleusSamplingFactor = data.nucleusSamplingFactor;
        this.presencePenalty = data.presencePenalty;
        this.frequencyPenalty = data.frequencyPenalty;
        this.temperature = data.temperature;

    }
}
